node/src/node_configurator/node_configurator_standard.rs

```diff
@@ -737,6 +736,295 @@ mod tests {
+            config_file
+                .write_all(b"payment-thresholds = \"3333|55|33|646|999|999\"\n")
+                .unwrap();
+            config_file
+                .write_all(b"scan-intervals = \"111|100|99\"\n")
+                .unwrap()
+        }
+    }
+
+    #[test]
+    fn server_initializer_collected_params_handle_config_file_from_environment_and_real_user_from_config_file_with_path_started_by_dot() {
+        running_test();
+        let home_dir = ensure_node_home_directory_exists( "node_configurator_standard","server_initializer_collected_params_handle_config_file_from_environment_and_real_user_from_config_file_with_path_started_by_dot");
+        let data_dir = &home_dir.join("data_dir");
+        let config_file_relative = File::create(PathBuf::from("./generated").join("config.toml")).unwrap();
+        fill_up_config_file(config_file_relative);
+        let env_vec_array = vec![
+            ("MASQ_CONFIG_FILE", "./generated/config.toml"),
+            #[cfg(not(target_os = "windows"))]
+            ("MASQ_REAL_USER", "9999:9999:booga"),
+        ];
+        env_vec_array.clone().into_iter()
+            .for_each (|(name, value)| std::env::set_var (name, value));
+        let args = ArgsBuilder::new();
+        let args_vec: Vec<String> = args.into();
+        let dir_wrapper = DirsWrapperMock::new()
+            .home_dir_result(Some(home_dir.clone()))
+            .data_dir_result(Some(data_dir.to_path_buf()));
+
+        let result = server_initializer_collected_params(&dir_wrapper, args_vec.as_slice());
+        let env_multiconfig = result.unwrap();
+
+        match env_multiconfig.is_user_specified("--data-directory") {
+            true => (),
+            false => {
+                assert_eq!(
+                    value_m!(env_multiconfig, "data-directory", String).unwrap(),
+                    "/home/wooga/data_dir/MASQ/polygon-mainnet".to_string()
+                )
+            }
+        }
+
+        match (env_multiconfig.is_user_specified("--config-file"), env_multiconfig.is_user_specified("--data-directory")) {
+            (true, true) => {
+                let pwd = PathBuf::from( value_m!(env_multiconfig, "data-directory", String).unwrap());
+                assert_eq!(
+                    value_m!(env_multiconfig, "config-file", String).unwrap(),
+                    pwd.join(PathBuf::from( "generated/config.toml")).to_string_lossy().to_string()
+                )
+            },
+            (true, false) => {
+                let pwd = current_dir().unwrap();
+                assert_eq!(
+                    value_m!(env_multiconfig, "config-file", String).unwrap(),
+                    pwd.join(PathBuf::from( "generated/config.toml")).to_string_lossy().to_string()
+                )
+            },
+            (_, _) => ()
+        }
+    }
+
+    #[test]
+    fn server_initializer_collected_params_handle_config_file_from_environment_and_real_user_from_config_file_with_data_directory() {
```

Comment:
"handleS"

The name stated like this isn't really clear can you restate it a bit?